### PR TITLE
Change `_Always_false` to `false` for functions

### PR DESCRIPTION
**Pull request by StephanTLavavej**
* Update the TRANSITION comment for `_Always_false`.
  + DevCom-10554332 VSO-1936351 was fixed by MSVC-PR-520085 on 2024-01-09, shipping in VS 2022 17.10 Preview 1.
  + Clang 17 started supporting this.
  + EDG isn't completely happy yet, see VSO-2016422. Functions work, but not structs/classes.
* Replace `_Always_false` usage in functions with `false`.
* Replace `always_false` test helpers in functions.
* Replace `AlwaysFalse` in `VSO_0000000_more_pair_tuple_sfinae` (thanks @frederick-vs-ja!).
* Cleanup: Fuse `static_assert`s in `_Move_only_function_call`.
  + This was attempting to generate a dedicated diagnostic for CTAD, but it wasn't working. (It could be fixed for Clang, but it would look weird, and it still wouldn't work for MSVC.) Just mention both scenarios.
  + Found while investigating struct usage, so this is semi-related.

### CODE REVIEW

**Comment by frederick-vs-ja** [comment]
Perhaps we should also remove `AlwaysFalse ` from `VSO_0000000_more_pair_tuple_sfinae`.

https://github.com/microsoft/STL/blob/9aca22477df4eed3222b4974746ee79129eb44e7/tests/std/tests/VSO_0000000_more_pair_tuple_sfinae/test.cpp#L646-L671

**Review by AlexGuteniev — APPROVED** [review]

File: tests/std/include/range_algorithm_support.hpp [hunk 1]
```diff
@@ -69,12 +69,12 @@ namespace test {
 
     template <class T>
     void operator&(T&&) {
-        STATIC_ASSERT(always_false<T>);
+        STATIC_ASSERT(false);
```
Comment (CaseyCarter) [inline tests/std/include/range_algorithm_support.hpp:72]: Should we also mark the definition of `always_false` on line 22 as `TRANSITION, VSO-2016422 (EDG)` to ensure it isn't missed when the bug gets fixed?

**Review by CaseyCarter — APPROVED** [review]

File: tests/std/include/range_algorithm_support.hpp (hunk 1 above)
Comment (CaseyCarter) [reply tests/std/include/range_algorithm_support.hpp:72]: I pushed a fix for this, and also annotated the definition of `always_false` in `tests/P0898R3_concepts`

Comment (StephanTLavavej) [reply tests/std/include/range_algorithm_support.hpp:72]: Great idea, thank you! :heart_eyes_cat:

**Comment by StephanTLavavej** [comment]
I'm mirroring this to the MSVC-internal repo - please notify me if any further changes are pushed.

**Comment by StephanTLavavej** [comment]
Azure Pipelines checks are green, but GitHub's UI isn't reflecting that. I'm going to need to resolve merge conflicts with #4588 so there's no point in forcing a rerun now.

**Pull request merged**